### ISSUE TITLE
Validator incorrectly mandating width and height in SVG - reopen

### ISSUE THREAD

**Issue by KDean-Dolphin** [validator/validator#2036]
### Summary

This is a reopen of #1988. The problem still exists for `foreignObject`, and may still exist for `svg` and `image`, as none of those were mentioned in the fix.

Thanks.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
See #1988.
```

### Checker error messages

```shell
Element “foreignObject” is missing required attribute “height”.
```

### Spec links/citations

See #1988.

### Steps to reproduce

See #1988.

### Expected behavior

See #1988.

### Actual behavior

See #1988.

**Comment by sideshowbarker** [comment]
Fixed in https://github.com/validator/validator/commit/9a55ab3f8fc84207649040bc23f61dcfbce3ebe4

**Comment by KDean-Dolphin** [comment]
@sideshowbarker Still a problem. This is the latest output, showing that it is picking up the recently deployed vnc-jar. If you want the HTML, email me at the address in my GitHub profile.

  $ java -jar "/home/runner/work/_actions/w3c/spec-prod/v2/node_modules/.pnpm/vnu-jar@26.2.5/node_modules/vnu-jar/build/dist/vnu.jar" --also-check-css index.html
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":147.1-147.31: info warning: The “type” attribute is unnecessary for JavaScript resources.
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":696.76-696.279: error: Discarding superfluous token “document” from value of attribute “role”. Browsers only process the first token found that is a defined ARIA non-abstract role.
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":698.10090-698.10114: error: Element “foreignObject” is missing required attribute “height”.

**Comment by sideshowbarker** [comment]
What version string does `java -jar vnu.jar --version` emit?

**Comment by KDean-Dolphin** [comment]
I don't have access to the GitHub workflow to tell you that, but you can see in the above that it's using vnu-jar@26.2.5. Previous runs of the workflow showed older versions in the command line string.

**Comment by sideshowbarker** [comment]
I've been away from computer so I've not yet been able to try to reproduce this yet. But what I can imagine being possible is that the vnu.jar that got used is an earlier build from today that got built before the relevant changes landed.

The 26.2.5 jar number doesn't uniquely identify the exact build tag of the jar. It's possible and common for the project to release multiple releases on the same day that all end up with the same release substring in the jar number. Running the jar with the `--version` option is the only way to identify the exact tag build tag it was built from.

**Comment by KDean-Dolphin** [comment]
The JAR file is downloaded dynamically. I reran the job just now, so whatever version the workflow got via `pnpm add vnu-jar` is what I'm running against:

```
$ pnpm add vnu-jar
+ vnu-jar 26.2.5
$ java -jar "/home/runner/work/_actions/w3c/spec-prod/v2/node_modules/.pnpm/vnu-jar@26.2.5/node_modules/vnu-jar/build/dist/vnu.jar" --also-check-css index.html
"file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":147.1-147.31: info warning: The “type” attribute is unnecessary for JavaScript resources.
"file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":696.76-696.279: error: Discarding superfluous token “document” from value of attribute “role”. Browsers only process the first token found that is a defined ARIA non-abstract role.
"file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":698.10090-698.10114: error: Element “foreignObject” is missing required attribute “height”.
```

**Comment by KDean-Dolphin** [comment]
@sideshowbarker Any word on this?

**Comment by sideshowbarker** [comment]
> @sideshowbarker Any word on this?

Have you tested with the latest npm release?

**Comment by KDean-Dolphin** [comment]
I'm afraid I don't understand your versioning process. This is the output of the job I just ran:

```shell
  $ pnpm add vnu-jar
  + vnu-jar 26.2.12
  $ java -jar "/home/runner/work/_actions/w3c/spec-prod/v2/node_modules/.pnpm/vnu-jar@26.2.12/node_modules/vnu-jar/build/dist/vnu.jar" --also-check-css index.html
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":147.1-147.31: info warning: The “type” attribute is unnecessary for JavaScript resources.
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":696.76-696.279: error: Discarding superfluous token “document” from value of attribute “role”. Browsers only process the first token found that is a defined ARIA non-abstract role.
    "file:/home/runner/work/vc-use-cases/vc-use-cases.w3c/index.html":698.10090-698.10114: error: Element “foreignObject” is missing required attribute “height”.
```

I'm clearly getting the latest vnu-jar (26.2.12). What's missing?

I took a look at the build tool itself and its `package.json` includes the following:

```json
	"pnpm": {
		"onlyBuiltDependencies": [
			"puppeteer"
		],
		"overrides": {
			"link-checker": "1.4.2",
			"reffy": "^15",
			"respec": "^35",
			"specberus": "^11",
			"vnu-jar": "^26",
			"webidl2": "^24"
		}
	}
```

The `ppm-lock.yaml` file is no different:

```yaml
overrides:
  link-checker: 1.4.2
  reffy: ^15
  respec: ^35
  specberus: ^11
  vnu-jar: ^26
  webidl2: ^24
```

**Comment by sideshowbarker** [comment]
OK, thanks for your patience and for the details. From that I managed to figure out that the jar-file release artifact in all recent releases has not been getting updated for more than a month now.


So in https://github.com/validator/validator/commit/f73f3283225f315a06c6cae54bc3d834560cf790 I fixed the cause of that. But a new npm with that fix — 26.2.13 — will not go out until tomorrow.

**Comment by KDean-Dolphin** [comment]
Much to my frustration, something else is going wrong with the toolchain, prior to the validator being called. I'll get back to you once I'm able to test.

In the meantime, can you update the online validator at https://validator.nu? It's on version 22.3.8, and should really track the releases.

**Comment by sideshowbarker** [comment]
> In the meantime, can you update the online validator at https://validator.nu?

No, I can’t, unfortunately. I don’t maintain it. Nobody nor anything should be using it any longer, really. Everyone and everything should instead be using https://validator.w3.org/nu/.

**Comment by KDean-Dolphin** [comment]
I'll update my bookmarks and pass on that information, thanks.

The "something else going wrong with the toolchain" resolved itself as mysteriously as it appeared, and the workflow job has now completed successfully. Thanks for helping me through this.
